source/extensions/access_loggers/grpc/tcp_config.cc

```diff
@@ -0,0 +1,51 @@
+#include "extensions/access_loggers/grpc/tcp_config.h"
+
+#include "envoy/config/accesslog/v2/als.pb.validate.h"
+#include "envoy/config/filter/accesslog/v2/accesslog.pb.validate.h"
+#include "envoy/registry/registry.h"
+#include "envoy/server/filter_config.h"
+
+#include "common/common/assert.h"
+#include "common/common/macros.h"
+#include "common/grpc/async_client_impl.h"
+#include "common/protobuf/protobuf.h"
+
+#include "extensions/access_loggers/grpc/config_utils.h"
+#include "extensions/access_loggers/grpc/grpc_access_log_proto_descriptors.h"
+#include "extensions/access_loggers/grpc/tcp_grpc_access_log_impl.h"
+#include "extensions/access_loggers/well_known_names.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace AccessLoggers {
+namespace TcpGrpc {
+
+AccessLog::InstanceSharedPtr
+TcpGrpcAccessLogFactory::createAccessLogInstance(const Protobuf::Message& config,
+                                                 AccessLog::FilterPtr&& filter,
+                                                 Server::Configuration::FactoryContext& context) {
+  GrpcCommon::validateProtoDescriptors();
+
+  const auto& proto_config =
+      MessageUtil::downcastAndValidate<const envoy::config::accesslog::v2::TcpGrpcAccessLogConfig&>(
+          config);
+
+  return std::make_shared<TcpGrpcAccessLog>(std::move(filter), proto_config, context.threadLocal(),
+                                            GrpcCommon::getGrpcAccessLoggerCacheSingleton(context));
+}
+
+ProtobufTypes::MessagePtr TcpGrpcAccessLogFactory::createEmptyConfigProto() {
+  return ProtobufTypes::MessagePtr{new envoy::config::accesslog::v2::TcpGrpcAccessLogConfig()};
```

Comment:
Can we use `std::make_unique` here instead of direct call of `new`, please?

Reply:
yep, fixed http as well